src/storage/hummock_sdk/src/lib.rs

```diff
@@ -266,3 +267,55 @@ pub fn version_checkpoint_path(root_dir: &str) -> String {
 pub fn version_checkpoint_dir(checkpoint_path: &str) -> String {
     checkpoint_path.trim_end_matches(|c| c != '/').to_string()
 }
+
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Default, Debug, PartialOrd, Ord)]
+pub struct EpochWithGap(u64);
+
+impl EpochWithGap {
+    pub fn new(epoch: u64, spill_offset: u64) -> Self {
+        #[cfg(not(feature = "enable_test_epoch"))]
+        {
+            let epoch_with_gap = epoch + spill_offset;
+            EpochWithGap(epoch_with_gap)
+        }
+        #[cfg(feature = "enable_test_epoch")]
+        {
+            let epoch_with_gap = epoch + spill_offset - spill_offset;
+            EpochWithGap(epoch_with_gap)
+        }
+    }
+
+    pub fn new_from_epoch(epoch: u64) -> Self {
+        #[cfg(not(feature = "enable_test_epoch"))]
```

Comment:
Better to move the sanity check to `new`.

Reply:
Seems not resolved yet.

Reply:
As we will construct `EpochWithGap::new(self.read_epoch, MAX_SPILL_TIMES)` to do the seek, I move the check in `new_from_epoch`.